### ISSUE TITLE
Run unsupported platforms anyways, to hopefully improve platform coverage

### ISSUE THREAD

**Issue by infinisil** [NixOS/ofborg#226]
When a build doesn't eval due to `unsupported platform`, ofborg could run the build anyways, to see if it would succeed, if this were an allowed platform. In the reporting comment, it should then mention that:

> Successfully built on darwin, even though it's not in meta.platforms

or

> Failed to build on darwin, in accordance with this platform not being in meta.platforms

(or something along those lines)

This could improve the platform coverage situation.

**Comment by infinisil** [comment]
Caught in the wild: https://github.com/NixOS/nixpkgs/pull/43754#issuecomment-415961136

**Comment by Enzime** [comment]
For that PR more specifically, I'm not a trusted user so it doesn't run Darwin builds.

Potentially relevant issue: https://github.com/NixOS/ofborg/issues/204

**Comment by infinisil** [comment]
@Enzime Yeah, the first build however was a non-attempted darwin one: https://github.com/NixOS/nixpkgs/pull/43754#issuecomment-406035129
